leader-election-impl/src/main/java/com/palantir/leader/LeadershipEvents.java

```diff
@@ -48,8 +48,11 @@
     private final Meter leaderPingReturnedFalse;
     private final Object[] contextArgs;
 
-    LeadershipEvents(TaggedMetricRegistry metrics, List<SafeArg<String>> contextArgs) {
-        Map<String, String> safeTags = safeTags(contextArgs);
+    LeadershipEvents(
+            TaggedMetricRegistry metrics,
```

Comment:
TaggedMetricRegistries are nice because of the:

```
void addMetrics(String safeTagName, String safeTagValue, TaggedMetricSet metrics)
```

Method, which basically allows for establishing metric hierarchies. Using this, this class would not have to deal with tags at all, because it would be passed a TaggedMetricRegistry that is scoped to the exact thing it needs, and ALL metrics created underneath it would be tagged correctly.

Reply:
Deregistration in the TimelockLeadershipMetrics should also hopefully become just a simple removal of the scoped TaggedMetricSet, so that code should be less error prone.

Reply:
so I'm using that above, I didn't do another level for reasons unbeknownst to me, perhaps I was worried about perf of multi level hierarchies, but I can give it a try.

Reply:
If it's not performant enough, we should fix this in the library and not workaround here.